lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -597,26 +634,58 @@ public static function get_merged_data( $origin = 'custom' ) {
 			_deprecated_argument( __FUNCTION__, '5.9.0' );
 		}
 
+		if ( ! in_array( $origin, WP_Theme_JSON_Gutenberg::VALID_ORIGINS, true ) ) {
+			$origin = 'custom';
+		}
+
+		// Map origins to block cache values.
+		$cache_map = array(
+			'default' => 'core',
+			'blocks'  => 'blocks',
+			'theme'   => 'theme',
+			'custom'  => 'user',
+		);
+
+		if (
+			null !== static::$merged[ $origin ]
+			&& static::has_same_registered_blocks( $cache_map[ $origin ] )
+			// Ensure theme supports data is fresh before returning cached data for theme and custom origins.
+			&& ( ! in_array( $origin, array( 'theme', 'custom' ), true ) || static::get_theme_supports_data() === static::$theme_support_data )
```

Comment:
I presume the logic above to convert unknown origins to `custom` was necessary for this line. Thinking we can do this instead:

> // `theme` and `custom` origins require access to theme supports,
> // so the cached merged data will only be used for those origins if the current supports are equal to the cached ones. 
> ( in_array( $origin, array( 'default', 'blocks' ), true ) || static::get_theme_supports_data() === static::$theme_support_data )